File: dcp-solutions/2025-02-03/solution.go
```markdown
package main

import (
	"container/list"
)

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func pruneTree(root *TreeNode) *TreeNode {
	var deleteList list.List
	deleteList.PushBack(root)
	for deleteList.Len() > 0 {
		node := deleteList.Front().Value.(*TreeNode)
		deleteList.Remove(deleteList.Front())

		if node.Left != nil && (node.Left.Val == 1 || pruneTree(node.Left) != nil) {
			node.Left = nil
			deleteList.PushBack(node.Left)
		}

		if node.Right != nil && (node.Right.Val == 1 || pruneTree(node.Right) != nil) {
			node.Right = nil
			deleteList.PushBack(node.Right)
		}
	}

	return root
}
```

File: dcp-solutions/2025-02-03/solution.php
```php
class Solution {
    function pruneTree($root) {
        if (!$root) return null;
        
        $root->left = $this->pruneTree($root->left);
        $root->right = $this->pruneTree($root->right);

        if ($root->val == 0 && !$root->left && !$root->right) {
            return null;
        }
        
        return $root;
    }
}
```